### PR TITLE
Update rustls

### PR DESCRIPTION
**Pull request by richardpringle**
Update rustls and surrounding dependencies

### CODE REVIEW

File: core/network/src/peer/mod.rs [hunk 1]
```diff
@@ -64,7 +64,6 @@ mod test {
 
         let join_handle = tokio::task::spawn(async move {
             let server_config = ServerConfig::builder()
-                .with_safe_defaults()
```
Comment (richardpringle) [inline core/network/src/peer/mod.rs:67]: https://docs.rs/rustls/latest/src/rustls/server/server_conn.rs.html#349-354

Looks like "safe defaults" are automatically applied to the builder

File: core/network/src/peer/outbound.rs [hunk 2]
```diff
@@ -54,72 +57,117 @@ impl Connector {
...
+            rustls::SignatureScheme::ECDSA_NISTP384_SHA384,
+            rustls::SignatureScheme::ECDSA_NISTP521_SHA512,
+            rustls::SignatureScheme::ED25519,
+            rustls::SignatureScheme::RSA_PSS_SHA256,
+            rustls::SignatureScheme::RSA_PSS_SHA384,
+            rustls::SignatureScheme::RSA_PSS_SHA512,
+            rustls::SignatureScheme::RSA_PKCS1_SHA256,
+            rustls::SignatureScheme::RSA_PKCS1_SHA384,
+            rustls::SignatureScheme::RSA_PKCS1_SHA512,
+        ]
```
Comment (richardpringle) [inline core/network/src/peer/outbound.rs:170]: @gyuho, should I limit this?

**Review by gyuho — APPROVED** [review]
lgtm

**Review by gyuho — APPROVED** [review]
Lgtm. Thanks 🙏

**Pull request merged**